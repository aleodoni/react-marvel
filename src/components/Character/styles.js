import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  justify-content: center;

  max-width: 1100px;
  width: 100%;
  margin: 0 auto;
`;

export const Thumb = styled.div`
  display: flex;

  img {
    width: 80px;
    height: 80px;
    border-radius: 50%;
    margin-right: 20px;
  }
`;

export const Name = styled.div`
  display: flex;

  width: 100%;
  color: #fff;
  font-size: 24px;
`;

export const Content = styled.div`
  display: flex;
  /* flex: 1; */
  justify-content: space-around;
  align-items: center;

  opacity: 0.8;
  transition: opacity 0.2s;

  &:hover {
    opacity: 1;
  }
  border-radius: 4px;
  padding: 15px;
  background: rgba(0, 0, 0, 0.5);
  width: 500px;
  margin: 5px;
  cursor: pointer;
`;
