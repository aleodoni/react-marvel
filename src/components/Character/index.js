import React from 'react';
import PropTypes from 'prop-types';

import { Container, Thumb, Name, Content } from './styles';

export default function Character({ url, name, handleClick }) {
  return (
    <Container>
      <Content onClick={handleClick}>
        <Thumb>
          <img src={url} alt="imagem" data-testid="character-img" />
        </Thumb>
        <Name data-testid="character-name">{name}</Name>
      </Content>
    </Container>
  );
}

Character.propTypes = {
  url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};
