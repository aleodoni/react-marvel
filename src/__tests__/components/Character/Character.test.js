import React from 'react';
import { render, cleanup, getByTestId } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Character from '../../../components/Character';

afterEach(cleanup);

it('renderiza corretamente', () => {
  function handleClick() {}

  const url = 'http://imagem.jpg';
  const name = 'name';

  const { container } = render(
    <Character url={url} name={name} handleClick={handleClick} />
  );

  expect(container).toMatchSnapshot();
});

it('testa atributos', () => {
  function handleClick() {}

  const url = 'http://imagem.jpg';
  const name = 'name';

  const { container } = render(
    <Character url={url} name={name} handleClick={handleClick} />
  );
  const characterImg = getByTestId(container, 'character-img');
  const characterName = getByTestId(container, 'character-name');

  expect(characterImg).toHaveAttribute('src', 'http://imagem.jpg');
  expect(characterName).toHaveTextContent('name');
});
