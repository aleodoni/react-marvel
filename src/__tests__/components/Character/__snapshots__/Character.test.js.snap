// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza corretamente 1`] = `
<div>
  <div
    class="sc-bdVaJa kuQaNn"
  >
    <div
      class="sc-bxivhb bYiJEU"
    >
      <div
        class="sc-bwzfXH dRLscd"
      >
        <img
          alt="imagem"
          data-testid="character-img"
          src="http://imagem.jpg"
        />
      </div>
      <div
        class="sc-htpNat cJsDH"
        data-testid="character-name"
      >
        name
      </div>
    </div>
  </div>
</div>
`;
